app/models/goal.py

```diff
@@ -1,5 +1,27 @@
-from sqlalchemy.orm import Mapped, mapped_column
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+from .task import Task
 from ..db import db
+from typing import List
 
 class Goal(db.Model):
     id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
+    title: Mapped[str]
+    tasks: Mapped[List["Task"]] = relationship("Task", back_populates="goal")
+
+    def to_dict(self):
+        goal_as_dict = {
+            "id": self.id,
+            "title": self.title,
+            "tasks": [task.to_dict() for task in self.tasks]  # Include tasks in the goal's dict
```

Comment:
We have an endpoint that will provide us with the same data that `goal_as_dict["tasks"]` will give us so we don't need to put it on the dictionary representation of the object. If I client has the information to retrieve a `Goal` (i.e. `goal.id`) then they have the information to get the same information from our endpoint.